File: xssfinder-executor-java/src/test/java/org/xssfinder/runner/TraversalResultTest.java
package org.xssfinder.runner;

import com.google.common.collect.ImmutableMap;
import org.junit.Test;

import java.util.Map;

import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class TraversalResultTest {
    @Test
    public void resultingPageAndAttackedIdsAreAvailable() {
        // given
        Object givenPage = mock(Object.class);
        Map<String, String> givenInputIdsToAttackIds = ImmutableMap.of("foo", "bar");
        TraversalResult result = new TraversalResult(givenPage, givenInputIdsToAttackIds);

        // when
        Object page = result.getPage();
        Map<String, String> inputIdsToAttackIds = result.getInputIdsToAttackIds();

        // then
        assertThat(page, is(givenPage));
        assertThat(inputIdsToAttackIds, is(givenInputIdsToAttackIds));
    }
}

File: xssfinder-php-executor/test/XssFinder/Runner/TraversalResultTest.php
<?php

namespace XssFinder\Runner;

use PHPUnit_Framework_TestCase;

class TraversalResultTest extends PHPUnit_Framework_TestCase
{
    public function testHoldsInputIdsToAttackIds()
    {
        // given
        $inputIdsToAttackIds = array('//some/thing' => '123');
        $traversalResult = new TraversalResult(new \stdClass(), $inputIdsToAttackIds);

        // when
        $ids = $traversalResult->getInputIdsToAttackIds();

        // then
        assertThat($ids, is($inputIdsToAttackIds));
    }

    public function testHoldsPage()
    {
        // given
        $mockPage = new \stdClass();
        $traversalResult = new TraversalResult($mockPage, array());

        // when
        $page = $traversalResult->getPage();

        // then
        assertThat($page, is($mockPage));
    }
}